packages/app/src/components/faucet/index.tsx

```diff
@@ -13,11 +14,17 @@ const Faucet: FC<props> = ({ symbol }) => {
   const handleClose = () => setShow(false);
   const handleShow = () => setShow(true);
 
-  const { send } = useContractFunction(contracts.livepeer.faucet, "request");
+  const location = useLocation();
+  const name = location.pathname.split("/")[2];
 
+  console.log(contracts[name]);
+  const { send } = useContractFunction(contracts[name].faucet, "request");
+
+  // TODO why is this needed
   const requestTokens = () => {
     send();
   };
+
```

Comment:
```suggestion
  const { send: requestTokens } = useContractFunction(contracts[name].faucet, "request");
```

Reply:
actually I tried that, and does not work, hence I added the TODO